examples/cli/readme.md

```diff
@@ -1 +1,619 @@
-placeholder
\ No newline at end of file
+# Learning the `benchit` CLI
```

Comment:
Very well written but maybe a bit long? Potentially we can divide this into a basic and one advanced tutorial.

Reply:
I'm open to this if there's a principled separation into chapters, and each chapter is its own markdown file. Suggestions on chapters?

Reply:
1. Getting Started (hello world, --all, --device, etc.)
2. Guided Model Discovery (--analyze-only, hashes, --max-depth, etc.)
3. Working with the Cache (--cache-dir, cache commands, etc.)

?

Reply:
I really like the chapters you created. That will make getting starting a lot less intimidating. 

Also, I would definitely recommend people to follow those chapters in that order.

Reply:
Done! There are now 4 ordered chapters. Check it out!

Reply:
Looking good!